server/market/market.go

```diff
@@ -587,6 +588,163 @@ func (m *Market) SubmitOrder(rec *orderRecord) error {
+			order:    lo,
+			epochIdx: 0,
+		},
+	}
+	m.sendToFeeds(sig)
+
+	match := order.Match{
+		Taker:    co,
+		Maker:    lo,
+		Quantity: lo.Remaining(),
+		Rate:     lo.Rate,
+		Epoch: order.EpochID{
+			Idx: uint64(epochIdx),
+			Dur: m.EpochDuration(),
+		},
+		FeeRateBase:  m.getFeeRate(m.Base(), m.baseFeeFetcher),
+		FeeRateQuote: m.getFeeRate(m.Quote(), m.quoteFeeFetcher),
+	}
+	m.storage.InsertMatch(&match)
+	makerMsg, takerMsg := matchNotifications(&match)
+	m.auth.Sign(makerMsg)
+	m.auth.Sign(takerMsg)
+	msgs := []msgjson.Signable{makerMsg, takerMsg}
+	req, err := msgjson.NewRequest(comms.NextID(), msgjson.MatchRoute, msgs)
+	if err != nil {
+		log.Errorf("Failed to create match reqeust: %v", err)
+	} else {
+		err = m.auth.Request(rec.order.User(), req, func(_ comms.Link, resp *msgjson.Message) {
+			m.processMatchAcksForCancel(rec.order.User(), resp)
+		})
+		if err != nil {
+			log.Errorf("Failed to send match reqeust: %v", err)
+		}
+	}
+
+	errChan <- sendResponseErr
+}
+
+// matchNotifications creates a pair of msgjson.Match from a match.
+func matchNotifications(match *order.Match) (makerMsg *msgjson.Match, takerMsg *msgjson.Match) {
+	stamp := encode.UnixMilliU(time.Now())
+	return &msgjson.Match{
+			OrderID:      idToBytes(match.Maker.ID()),
+			MatchID:      idToBytes(match.ID()),
+			Quantity:     match.Quantity,
+			Rate:         match.Rate,
+			Address:      order.ExtractAddress(match.Taker),
+			ServerTime:   stamp,
+			FeeRateBase:  match.FeeRateBase,
+			FeeRateQuote: match.FeeRateQuote,
+			Side:         uint8(order.Maker),
+		}, &msgjson.Match{
+			OrderID:      idToBytes(match.Taker.ID()),
+			MatchID:      idToBytes(match.ID()),
+			Quantity:     match.Quantity,
+			Rate:         match.Rate,
+			Address:      order.ExtractAddress(match.Maker),
+			ServerTime:   stamp,
+			FeeRateBase:  match.FeeRateBase,
+			FeeRateQuote: match.FeeRateQuote,
+			Side:         uint8(order.Taker),
+		}
+}
+
+// processMatchAcksForCancel is called when receiving a response to a match
+// request for a cancel order. Nothing is done other than logging and verifying
+// that the response is in the correct format.
+//
+// This is currently only used for cancel orders that happen while the market is
+// suspended, but may be later used for all cancel orders.
+func (m *Market) processMatchAcksForCancel(user account.AccountID, msg *msgjson.Message) {
+	var acks []msgjson.Acknowledgement
+	err := msg.UnmarshalResult(&acks)
+	if err != nil {
+		m.respondError(msg.ID, user, msgjson.RPCParseError,
+			"error parsing match request acknowledgment")
+		return
+	}
+	// The acknowledgment for both the taker and maker should come from the same user
+	expectedNumAcks := 2
+	if len(acks) != expectedNumAcks {
+		m.respondError(msg.ID, user, msgjson.AckCountError,
+			fmt.Sprintf("expected %d acknowledgements, got %d", len(acks), expectedNumAcks))
+		return
+	}
+	log.Debugf("processMatchAcksForCancel: 'match' ack recieived from %v", user)
```

Comment:
recieived

Reply:
This word is my nemesis.

Reply:
Just remember "i before e except after c"

I have to repeat this to myself about 10 times a day.

Reply:
I just put the i on both sides because I couldn't decide haha.
I'll keep repeating it though and it probably won't ever happen again :)

Reply:
Make up any mnemonic that is easy to remember e.g. "a weighty sovereign reindeer receives forfeit caffeine".